plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/expression/SparkExpressionBuilder.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.expression;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static java.lang.String.format;
+import static java.util.HexFormat.isHexDigit;
+
+public class SparkExpressionBuilder
+        extends SparkExpressionBaseVisitor<Object>
+{
+    private static final char SPARK_ESCAPE_CHARACTER = '\\';
```

Comment:
we're in a Spark expression related class, so no need for "SPARK_" in name

however, it would be nice to indicate it's unicode escape for string literals (and not eg escape in LIKE patterns)

`STRING_LITERAL_ESCAPE_CHARACTER`